skimage/filters/ridges.py

```diff
@@ -105,9 +107,139 @@ def _check_sigmas(sigmas):
+    >>> square = np.zeros((30, 30))
+    >>> square[15, 15] = 1
+    >>> Hrr, Hrc, Hcc = hessian_matrix_with_Gaussian(square, sigma=1)
+    >>> import matplotlib.pyplot as plt
+    >>> plt.imshow(Hcc)
+    >>> plt.colorbar()
+    >>> plt.show()
+    """
+    image = img_as_float(image)
+    float_dtype = _supported_float_type(image.dtype)
+    image = image.astype(float_dtype, copy=False)
+
+    H_elems = []
+    idx = np.arange(image.ndim)
+    # The derivative of an image I convolved with a Gaussian G is
+    #       (d/dx_i)[I*G]
+    # where * indicates a convolution. The distributive property
+    # of derivatives and convolutions allows us to restate this as
+    #        I * dG/dx_i
+    # that is, the convolution of I with the derivative of a Gaussian.
+    # We need to call scipy.ndimage.gaussian_filter with the argument
+    # "order" which indicates the derivative order in the respective
+    # directions, where 0 = just Gaussian smoothing
+    #                   1 = convolve with first derivative of Gaussian
+    #                   etc.
+    # so supplying order=[2, 0] computes the 2nd Gaussian derivative in
+    # the first direction, and just smoothes the field in the second
+    # direction. That corresponds to the lower-right element of the
+    # Hessian matrix, Hcc. This is why below we will call the array
+    # deriv_step{1/2}[::-1] thus in reverse order, because the image
+    # will be in coordinates [(z,)y,x], but we need the Hessian in
+    # order [d^2/dx^2, d^2/(dx*dy), ...], so in reverse order.
+    for deriv_dirs in itertools.combinations_with_replacement(idx, 2):
+        # E.g., for idx=[0, 1] we get deriv_dirs=[0, 0]; [0, 1]; [1, 1]
+
+        deriv_step1 = 1 * (idx == deriv_dirs[0])
+        deriv_step2 = 1 * (idx == deriv_dirs[1])
+        # E.g., for deriv_dirs=[0, 0] we get deriv_step1=[1, 0]
+        #                                and deriv_step2=[1, 0]
+        #       for deriv_dirs=[0, 1] we get deriv_step1=[1, 0]
+        #                                and deriv_step2=[0, 1]
+        #       etc., expressing the two successive derivative
+        #             operations in the Hessian
+
+        if order == 'rc':
+            deriv_step1 = deriv_step1[::-1]
+            deriv_step2 = deriv_step2[::-1]
+            # For, e.g., deriv_order=[2, 0], we want the second
+            # derivative in the "horizontal"/"row" direction, and
+            # just Gaussian smoothing in the "vertical"/"column"
+            # direction. To do that on an array, we need to
+            # differentiate as [0, 2], because the first direction
+            # is the vertical direction, and the second the horizontal
+            # direction. Hence, we reverse the list order.
+
+        # Apply two successive Gaussian filter operations, as per detailed in
+        # https://dsp.stackexchange.com/questions/78280/are-scipy-second-order-gaussian-derivatives-correct
+        if np.all(np.array(sigma)>1):
```

Comment:
```suggestion
        if np.all(np.asarray(sigma) > 1):
```
This avoids a copy if sigma is already an `ndarray`.